### PR TITLE
reflect-cpp: add option for using with msgpack

### PR DESCRIPTION
**Pull request by uyha**
Specify library name and version:  **lib/1.0**

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by RubenRBS — COMMENTED** [review]
Hi @uyha thanks a lot for taking the time to create the PR. 

This being a header-only library, asking for compilation logs would not be too useful - could you instead provide an example program using this library with msgpack enabled that makes use of the new available features thanks to the new dependency? Thanks!

**Comment by uyha** [comment]
Do you mean add a test that uses the msgpack feature?

**Comment by RubenRBS** [comment]
yes, that'd work :)
Let me know if you need any help adding it

File: recipes/reflect-cpp/all/conanfile.py [hunk 1]
```diff
@@ -56,6 +58,8 @@ def requirements(self):
             self.requires("flatbuffers/23.5.26", transitive_headers=True)
         if self.options.with_yaml:
             self.requires("yaml-cpp/0.8.0", transitive_headers=True)
+        if self.options.with_msgpack:
+            self.requires("msgpack-cxx/6.1.1", transitive_headers=True)
```
Comment (jcar87) [inline recipes/reflect-cpp/all/conanfile.py:62]: `reflect-cpp` should require `msgpack-c`, not `msgpack-cxx`, as per readme here:

https://github.com/getml/reflect-cpp/blob/0342e5c9a224039146a9f4261ac3cb695457422a/README.md#L37-L38

**Comment by jcar87** [comment]
I've amended the PR to use the correct `msgpack-c` library, and amended the test so that this functionality is tested, as per the reflect-cpp readme.

@uyha, thanks for your contribution - please be mindful that for PRs that introduce features that are not tested by CI (such as an optional feature), we _do_ require that it is tested locally first, and it would appear this was not the case. We'll try to do it ourselves where possible - but otherwise we run the risk of introducing broken options, and it can only delay the PR being merged if the changes are useful.


```
(venv) root@079fdee7cb8c:/reflect-cpp/all/test_package# conan test . reflect-cpp/0.10.0  -s compiler.cppstd=20 -o "*:with_msgpack=True"

======== Input profiles ========
Profile host:
[settings]
arch=armv8
build_type=Release
compiler=gcc
compiler.cppstd=20
compiler.libcxx=libstdc++11
compiler.version=13
os=Linux
[options]
*:with_msgpack=True

Profile build:
[settings]
arch=armv8
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=13
os=Linux


======== Launching test_package ========

======== Computing dependency graph ========
Graph root
    reflect-cpp/0.10.0 (test package): /reflect-cpp/all/test_package/conanfile.py
Requirements
    msgpack-c/6.0.0#769a7b807a4b0f758f24755361069c68 - Cache
    reflect-cpp/0.10.0#dc114ad4bed2f24030d91807dc823210 - Cache

======== Computing necessary packages ========
Requirements
    msgpack-c/6.0.0#769a7b807a4b0f758f24755361069c68:258056170a5add18e564ac86c77027634e725af3#8b1fe2818466b69415b3668971c3575d - Cache
    reflect-cpp/0.10.0#dc114ad4bed2f24030d91807dc823210:da39a3ee5e6b4b0d3255bfef95601890afd80709#426c762de577fa544c490fdc098ca730 - Cache

======== Installing packages ========
msgpack-c/6.0.0: Already installed! (1 of 2)
reflect-cpp/0.10.0: Already installed! (2 of 2)
WARN: deprecated: Usage of deprecated Conan 1.X features that will be removed in Conan 2.X:
WARN: deprecated:     'cpp_info.names' used in: msgpack-c/6.0.0
WARN: deprecated:     'cpp_info.build_modules' used in: msgpack-c/6.0.0

======== Testing the package ========
Removing previously existing 'test_package' build folder: /reflect-cpp/all/test_package/build/gcc-13-armv8-20-release
reflect-cpp/0.10.0 (test package): Test package build: build/gcc-13-armv8-20-release
reflect-cpp/0.10.0 (test package): Test package build folder: /reflect-cpp/all/test_package/build/gcc-13-armv8-20-release
reflect-cpp/0.10.0 (test package): Writing generators to /reflect-cpp/all/test_package/build/gcc-13-armv8-20-release/generators
reflect-cpp/0.10.0 (test package): Generator 'CMakeDeps' calling 'generate()'
reflect-cpp/0.10.0 (test package): CMakeDeps necessary find_package() and targets for your CMakeLists.txt
    find_package(reflect-cpp)
    target_link_libraries(... reflect-cpp::reflect-cpp)
reflect-cpp/0.10.0 (test package): Generator 'VirtualRunEnv' calling 'generate()'
reflect-cpp/0.10.0 (test package): Calling generate()
reflect-cpp/0.10.0 (test package): Generators folder: /reflect-cpp/all/test_package/build/gcc-13-armv8-20-release/generators
reflect-cpp/0.10.0 (test package): CMakeToolchain generated: conan_toolchain.cmake
reflect-cpp/0.10.0 (test package): CMakeToolchain generated: /reflect-cpp/all/test_package/build/gcc-13-armv8-20-release/generators/CMakePresets.json
reflect-cpp/0.10.0 (test package): CMakeToolchain generated: /reflect-cpp/all/test_package/CMakeUserPresets.json
reflect-cpp/0.10.0 (test package): Generating aggregated env files
reflect-cpp/0.10.0 (test package): Generated aggregated env files: ['conanrun.sh', 'conanbuild.sh']

======== Testing the package: Building ========
reflect-cpp/0.10.0 (test package): Calling build()
reflect-cpp/0.10.0 (test package): Running CMake.configure()
reflect-cpp/0.10.0 (test package): RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/reflect-cpp/all/test_package" -DCONAN_TEST_WITH_MSGPACK="ON" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/reflect-cpp/all/test_package"
-- Using Conan toolchain: /reflect-cpp/all/test_package/build/gcc-13-armv8-20-release/generators/conan_toolchain.cmake
-- Conan toolchain: C++ Standard 20 with extensions OFF
-- The CXX compiler identification is GNU 13.2.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: Target declared 'reflect-cpp::reflect-cpp'
-- Conan: Target declared 'msgpack-c'
-- Configuring done (0.2s)
-- Generating done (0.0s)
-- Build files have been written to: /reflect-cpp/all/test_package/build/gcc-13-armv8-20-release

reflect-cpp/0.10.0 (test package): Running CMake.build()
reflect-cpp/0.10.0 (test package): RUN: cmake --build "/reflect-cpp/all/test_package/build/gcc-13-armv8-20-release" -- -j10
[ 50%] Building CXX object CMakeFiles/test_package.dir/test_package.cpp.o
[100%] Linking CXX executable test_package
[100%] Built target test_package


======== Testing the package: Executing test ========
reflect-cpp/0.10.0 (test package): Running test()
reflect-cpp/0.10.0 (test package): RUN: ./test_package
msgpack test: ��x�name�test_package
reflect-cpp test successful
```

**Review by RubenRBS — APPROVED** [review]

**Comment by RubenRBS** [comment]
Just as a tip for next time: `header-library` packages already infer `transitive_headers=True` for all their dependencies, it's not needed to explicitly set them :)

**Review by jcar87 — APPROVED** [review]

**Pull request merged**